src/main/java/wolox/training/models/Book.java

```diff
@@ -53,6 +57,62 @@
 
     @Column(nullable = false)
     @ManyToMany(mappedBy = "books")
+    @JsonIgnore
     private List<User> users;
 
+    public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre,String.format(Constants.MESSAGE_CHECK_IS_NULL,"genre"));
+        Preconditions.checkArgument(!genre.isEmpty(),String.format(Constants.MESSAGE_CHECK_IS_EMPTY,"genre"));
```

Comment:
This is good, but you can simplify to a single line. Here's a tip, the package `com.google.commons.base` has the `Strings` class with utils methods like `isNullOrEmpty`. Additionally, you can use static import to reduce the verbosity
```suggestion
        checkArgument(!Strings.isNullOrEmpty(genre), Strings.format(Constants.MESSAGE_CHECK_IS_EMPTY,"genre"));
```

Reply:
I almost forget it, remember unify the messages to one constant😛